### PR TITLE
Add a server greeting which is displayed to connecting user

### PR DESCRIPTION
**Pull request by delebota**
Adds a setting in config.ini to specify a message to display to players when they connect.

Somewhat addresses #4193 as an admin cannot change it without restarting the server. If we want to make is modifiable from in-game I can look into it.

### CODE REVIEW

File: src/network/network.cpp [hunk 1]
```diff
@@ -1947,6 +1951,14 @@ void Network::Client_Handle_GAMEINFO(NetworkConnection& connection, NetworkPacke
 		ServerProviderWebsite = json_stdstring_value(json_object_get(jsonProvider, "website"));
 	}
 	json_decref(root);
+
+	// Display server greeting if one exists
+	const char* greeting = network_get_server_greeting();
+	if (strcmp(greeting, "") != 0) {
+		char greeting_formatted[256];
```
Comment (janisozaur) [inline src/network/network.cpp]: Some hosts may choose to provide a longer message than this limit. Few bytes will be also taken by formatting, you should limit the user-defined greeting to something that would fit, say 240 chars and provide warning to user if this gets truncated. See `safe_str*` functions, they already do that.

**Comment by delebota** [comment]
Ok, looks like it goes off the screen around character 220. Is there a way to break it down into multiple lines? Otherwise we would want to limit it to around 200 characters. It looks like long chat messages don't get broken up into multiple lines either.

**Comment by IntelOrca** [comment]
Wrapping the chat should be fixed separately, in fact other people have made branches to make chat a window but they were never finished.

**Comment by pizza2004** [comment]
If you want I still have all of the changes necessary to make chat wrap.  Nubbie finished that stuff pretty well before he tried to move it into its own window and problems were encountered, but at the very least those commits are pretty nice.  I can certainly pull the two window commits off and fix merge commits and rebase for that functionality.

**Comment by joshbgosh10592** [comment]
For the server gretting, please to make it editable from within the game. That way the admin can go afk and update the greeting letting new users know.

**Pull request merged**